webpackfile.js

```diff
@@ -1,22 +1,50 @@
-const { resolve } = require('path');
+const { resolve, join } = require('path');
+const webpack = require('webpack');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
-const HtmlWebpackRootElementPlugin = require('html-webpack-root-element-plugin');
+const HtmlWebpackRootPlugin = require('html-webpack-root-plugin');
+const TSConfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
+
+/**
+ * NestJs uses a custom wrapper around require() for allows it to show a
+ * warning when some extra package needs to be installed. This causes problems
+ * with webpack, so we're blacklisting packages we're not using with the
+ * IgnorePlugin below.
+ *
+ * To de-blacklist a package, just remove it from this array.
+ */
+const nestBlacklist = [
+  '^class-validator$',
+  '^class-transformer$',
+  '^cache-manager$',
+  '^@nestjs/microservices$',
+  // packages below are required from microservices
+  '^amqp-connection-manager$',
+  '^amqplib$',
+  '^grpc$',
+  '^mqtt$',
+  '^nats$',
+  '^redis$',
+];
 
 const client = {
   name: 'client',
-  entry: ['./src/client/index.js'],
+  mode: 'production',
```

Comment:
I think it's fine since we're not using this webpack file for development. `'production'` is the default but webpack will complain if you don't specify one.

Reply:
Fair enough